packages/stk/stk_unit_tests/stk_topology/topology_test_utils.hpp

```diff
@@ -322,12 +322,15 @@ inline void check_permutation_nodes(stk::topology topology, std::vector<std::vec
   }
 }
 
+constexpr unsigned MAX_NODES_PER_ELEM = 100;
+
 template <unsigned MAX_NODES>
 STK_INLINE_FUNCTION
 void check_side_nodes_ngp(stk::topology topology, unsigned gold_side_node_ordinals[][MAX_NODES])
 {
-  constexpr unsigned MAX_NODES_PER_ELEM = 100;
   stk::mesh::Entity allElemNodes[MAX_NODES_PER_ELEM];
+  NGP_EXPECT_TRUE(topology.num_nodes() < MAX_NODES_PER_ELEM);
```

Comment:
Excellent, thanks!

Reply:
Just one suggestion.  You might consider an inlined function that checks this this and calls this in all three places.  Actually it looks like the code:

```
  stk::mesh::Entity allElemNodes[MAX_NODES_PER_ELEM];
  NGP_EXPECT_TRUE(topology.num_nodes() < MAX_NODES_PER_ELEM);
```

is 100% duplicated in these three places.   Would be easy to define a macro to reduce duplication like:

```
#define STK_DEFINE_ELEMENT_NODES_ARRAY(TOPOLOGY, ARRAY_NAME) \
  stk::mesh::Entity ARRAY_NAME[MAX_NODES_PER_ELEM]; \
  NGP_EXPECT_TRUE(TOPOLOGY.num_nodes() < MAX_NODES_PER_ELEM)
```

and then call it like:

```
  STK_DEFINE_ELEMENT_NODES_ARRAY(topology, allElemNodes);
```

Less duplication.  Less boilerplate.